core/src/main/java/org/apache/iceberg/BaseFile.java

```diff
@@ -0,0 +1,389 @@
+        return lowerBounds;
+      case 10:
+        return upperBounds;
+      case 11:
+        return keyMetadata != null ? ByteBuffer.wrap(keyMetadata) : null;
+      case 12:
+        return splitOffsets;
+      default:
+        throw new UnsupportedOperationException("Unknown field ordinal: " + pos);
+    }
+  }
+
+  @Override
+  public <T> T get(int pos, Class<T> javaClass) {
+    return javaClass.cast(get(pos));
+  }
+
+  @Override
+  public int size() {
+    return DataFile.getType(EMPTY_STRUCT_TYPE).fields().size();
+  }
+
+  public FileContent content() {
+    return content;
+  }
+
+  public CharSequence path() {
+    return filePath;
+  }
+
+  public FileFormat format() {
+    return format;
+  }
+
+  public StructLike partition() {
+    return partitionData;
+  }
+
+  public long recordCount() {
+    return recordCount;
+  }
+
+  public long fileSizeInBytes() {
+    return fileSizeInBytes;
+  }
+
+  public Map<Integer, Long> columnSizes() {
+    return columnSizes;
+  }
+
+  public Map<Integer, Long> valueCounts() {
+    return valueCounts;
+  }
+
+  public Map<Integer, Long> nullValueCounts() {
+    return nullValueCounts;
+  }
+
+  public Map<Integer, ByteBuffer> lowerBounds() {
+    return lowerBounds;
+  }
+
+  public Map<Integer, ByteBuffer> upperBounds() {
+    return upperBounds;
+  }
+
+  public ByteBuffer keyMetadata() {
+    return keyMetadata != null ? ByteBuffer.wrap(keyMetadata) : null;
+  }
+
+  public List<Long> splitOffsets() {
+    return splitOffsets;
+  }
+
+  private static <K, V> Map<K, V> copy(Map<K, V> map) {
+    if (map != null) {
+      Map<K, V> copy = Maps.newHashMapWithExpectedSize(map.size());
+      copy.putAll(map);
+      return Collections.unmodifiableMap(copy);
+    }
+    return null;
+  }
+
+  private static <E> List<E> copy(List<E> list) {
+    if (list != null) {
+      List<E> copy = Lists.newArrayListWithExpectedSize(list.size());
+      copy.addAll(list);
+      return Collections.unmodifiableList(copy);
+    }
+    return null;
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this)
+        .add("file_path", filePath)
+        .add("file_format", format)
+        .add("partition", partitionData)
+        .add("record_count", recordCount)
+        .add("file_size_in_bytes", fileSizeInBytes)
+        .add("column_sizes", columnSizes)
+        .add("value_counts", valueCounts)
+        .add("null_value_counts", nullValueCounts)
+        .add("lower_bounds", lowerBounds)
+        .add("upper_bounds", upperBounds)
+        .add("key_metadata", keyMetadata == null ? "null" : "(redacted)")
+        .add("split_offsets", splitOffsets == null ? "null" : splitOffsets)
+        .toString();
+  }
+
+  private static org.apache.avro.Schema getAvroSchema(Types.StructType partitionType) {
+    Types.StructType type = DataFile.getType(partitionType);
+    return AvroSchemaUtil.convert(type, ImmutableMap.of(
+        type, GenericDataFile.class.getName(),
```

Comment:
Do we keep `GenericDataFile` on purpose here?

Reply:
This is used by getSchema, which should probably be overridden. It doesn't matter for correctness, but it would be nice to use the right record name.

Reply:
Moving this into subclasses.